org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/util/JRTUtil.java

```diff
@@ -147,21 +148,9 @@ public static JrtFileSystem getJrtSystem(File image, String release) throws IOEx
 		if (release != null && !jdk.sameRelease(release)) {
 			key = key + "|" + release; //$NON-NLS-1$
 		}
-		try {
-			JrtFileSystem system = images.computeIfAbsent(key, x -> {
-				try {
-					return JrtFileSystem.getNewJrtFileSystem(jdk, release);
-				} catch (IOException e) {
-					// Needs better error handling downstream? But for now, make sure
-					// a dummy JrtFileSystem is not created.
-					String errorMessage = "Error: failed to create JrtFileSystem from " + image; //$NON-NLS-1$
-					throw new RuntimeIOException(errorMessage, e);
-				}
-			});
-			return system;
-		} catch (RuntimeIOException e) {
-				throw e.getCause();
-		}
+		JrtFileSystemReference reference = images.computeIfAbsent(key, x -> new JrtFileSystemReference(jdk,release));
```

Comment:
I think the key should be made from the `Jdk` itself since a normalized file name (toJdkHome) is stored in there.